contracts/src/GaspToken.sol

```diff
@@ -1,30 +1,126 @@
-// SPDX-License-Identifier: MIT
-pragma solidity ^0.8.12;
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.13;
 
-import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {IGaspToken} from "./interfaces/IGaspToken.sol";
 
-contract MintableERC20 is ERC20 {
-    address public owner;
-    constructor(string memory name_, string memory symbol_) ERC20(name_, symbol_) {
-        owner = msg.sender;
-        _mint(msg.sender, 1000000000 * 10 ** decimals());
+contract GaspToken is Context, Ownable, ERC20, IGaspToken {
+    uint256 private constant _TOTAL_SUPPLY = 1_000_000_000 * 10 ** 18;
+    string private constant _NAME = "GASP";
+    string private constant _SYMBOL = "GASP";
+
+    mapping(address => bool) public override whitelist;
+    address public override uniswapPool;
+    bool public override allowTransfers;
+
+    modifier isWhitelisted(address[2] memory addresses, bytes4 selector) {
+        if (!allowTransfers) {
+            if (!whitelist[addresses[0]] && !whitelist[addresses[1]] || addresses[1] == uniswapPool) {
+                revert OperationForbidden(selector);
+            }
+        }
+        _;
     }
 
-    function mint(address account, uint256 amount) public {
-      require((owner == msg.sender), "Only one who deployed contract can mint tokens");
-      _mint(account, amount);
+    constructor(address l1Council_) Ownable() ERC20(_NAME, _SYMBOL) {
+        if (l1Council_ == address(0)) {
+            revert ZeroL1Council();
+        }
+
+        _transferOwnership(l1Council_);
+        _mint(l1Council_, _TOTAL_SUPPLY);
+
+        whitelist[l1Council_] = true;
     }
-}
 
+    function setUniswapPool(address uniswapPool_) external override onlyOwner {
```

Comment:
Why is UniswapPool special here?
I didn't find anything about it in specs...

Reply:
I will send you the latest spec.